go/arrow/array/numeric.gen.go

```diff
@@ -817,6 +1045,22 @@ func (a *Time32) setData(data *Data) {
 	}
 }
 
+func (a *Time32) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+	return a.values[i].ToTime(a.DataType().(*arrow.Time32Type).Unit).Format("15:04:05.999999999")
```

Comment:
since it is a different type, it is probably easy to format this to only the precision required?

Reply:
using the `9`'s there will drop any trailing zeros so that it will only format it to the precision required to represent the value.